packages/openapi3/src/schema-emitter.ts

```diff
@@ -566,7 +604,16 @@ export class OpenAPI3SchemaEmitterBase<
   scalarDeclaration(scalar: Scalar, name: string): EmitterOutput<Schema> {
     const isStd = isStdType(this.emitter.getProgram(), scalar);
     const schema = this.#getSchemaForScalar(scalar);
-    const baseName = getOpenAPITypeName(this.emitter.getProgram(), scalar, this.#typeNameOptions());
+    let baseName = getOpenAPITypeName(this.emitter.getProgram(), scalar, this.#typeNameOptions());
+
+    ensureValidComponentFixedFieldKey(
+      this.emitter.getProgram(),
+      scalar,
+      () => scalar.name,
+      (newKey) => (scalar.name = newKey),
```

Comment:
you cannot do that this is mutating the program which is then potentially breaking all other emitters running at the same time.

Can't this logic just be done in `createDeclaration` which already has the name and handle the declaration registration?